airflow/jobs/scheduler_job.py

```diff
@@ -48,7 +49,7 @@
 from airflow.stats import Stats
 from airflow.ti_deps.dependencies_states import EXECUTION_STATES
 from airflow.utils import timezone
-from airflow.utils.callback_requests import DagCallbackRequest, TaskCallbackRequest
+from airflow.utils.callback_requests import DagCallbackRequest, SlaCallbackRequest, TaskCallbackRequest
```

Comment:
Let's move this out of `airflow.utils` and in to `airflow.callbacks` as utils is a huge messy dumping ground.

Reply:
If for some reason the DagFileProcessorManager fails, it restarted by heartbeat_manager method:
https://github.com/apache/airflow/blob/main/airflow/dag_processing/manager.py#L311

which recreates the pipes:
https://github.com/apache/airflow/blob/main/airflow/dag_processing/manager.py#L137